xdsl/dialects/onnx.py

```diff
@@ -26,6 +26,54 @@
 from xdsl.utils.exceptions import VerifyException
 
 
+def verify_shapes(
+    lhs_type: TensorType[Attribute] | list[int],
+    rhs_type: TensorType[Attribute] | list[int],
+    res_type: TensorType[Attribute],
+) -> None:
+    """
+    https://github.com/onnx/onnx/blob/main/docs/Broadcasting.md
+    """
+
+    # Check that the arguments are broadcastable (using Numpy semantics) and that the result type is correct.
+    res_shape: list[int] = []
+    if isinstance(lhs_type, TensorType):
+        lhs_shape = lhs_type.get_shape()
+    else:
+        lhs_shape = lhs_type
```

Comment:
```suggestion
    lhs_shape = lhs_type
    if isinstance(lhs_type, TensorType):
        lhs_shape = lhs_type.get_shape()
```

or even see if a ternary operator would make this more readable/fewer lines?

Same below.